lib/Doctrine/Common/Annotations/AnnotationReader.php

```diff
@@ -191,7 +191,24 @@ public function getClassAnnotations(ReflectionClass $class)
         $this->parser->setImports($this->getClassImports($class));
         $this->parser->setIgnoredAnnotationNames($this->getIgnoredAnnotationNames($class));
 
-        return $this->parser->parse($class->getDocComment(), 'class ' . $class->getName());
+        $annotations = $this->parser->parse($class->getDocComment(), 'class ' . $class->getName());
+        
+        if (!method_exists($class, 'getTraits')) {
+            return $annotations;
+        }
+        
+        $ignoredAnnotationNames = array_map('get_class', $annotations);
+        foreach($class->getTraits() as $trait) {
+            $this->parser->setImports($this->getClassImports($trait));
+            foreach($this->parser->parse($trait->getDocComment(), 'trait ' . $trait->getName()) as $annotation){
```

Comment:
What about methods imported from the trait (but replaced by the implementing class)?